src/test/java/no/unit/alma/acquisitions/AlmaPOLineServiceTest.java

```diff
@@ -0,0 +1,275 @@
+                "should have failed without poline_number.");
+    }
+
+    @Test
+    void testDeletePoline() {
+        when(mockAlmaApiClient.getWebTarget()).thenReturn(webTarget);
+        when(webTarget.path(anyString())).thenReturn(polineTarget);
+        when(polineTarget.path(anyString())).thenReturn(polineTarget);
+        when(polineTarget.queryParam(anyString(), any())).thenReturn(polineTarget);
+        when(polineTarget.request()).thenReturn(builder);
+        when(builder.accept(anyString())).thenReturn(builder);
+        when(builder.buildDelete()).thenReturn(invocation);
+        when(invocation.invoke()).thenReturn(response);
+        doNothing().when(response).close();
+
+        AlmaPOLineService almaPOLineService = new AlmaPOLineService(mockAlmaApiClient);
+        PoLine mockPoline = new PoLine();
+        mockPoline.setNumber(TEST_ID);
+        almaPOLineService.deletePoline(mockPoline);
+
+        mockPoline.setNumber("");
+        assertThrows(IllegalArgumentException.class, () -> almaPOLineService.deletePoline(mockPoline),
+                "should have failed without poline_number.");
+    }
+
+    @Test
+    void readAllActive() {
+        when(mockAlmaApiClient.getWebTarget()).thenReturn(acqTarget);
+        when(acqTarget.path(anyString())).thenReturn(polineTarget);
+        when(polineTarget.path(anyString())).thenReturn(polineTarget);
+        when(polineTarget.queryParam(anyString(), any())).thenReturn(polineTarget);
+        when(polineTarget.request()).thenReturn(builder);
+        when(builder.accept(anyString())).thenReturn(builder);
+        when(builder.buildGet()).thenReturn(invocation);
+        PoLines mockPolines = new PoLines();
+        mockPolines.setTotalRecordCount(Integer.valueOf(152));
+        when(invocation.invoke((Class<Object>) any())).thenReturn(mockPolines);
+
+        AlmaPOLineService almaPOLineService = new AlmaPOLineService(mockAlmaApiClient);
+        List<PoLines> poLines = almaPOLineService.readAllActive();
+        assertEquals(4, poLines.size());
+    }
+
+    @Test
+    void queryPoLines() {
+        when(mockAlmaApiClient.getWebTarget()).thenReturn(acqTarget);
+        when(acqTarget.path(anyString())).thenReturn(polineTarget);
+        when(polineTarget.path(anyString())).thenReturn(polineTarget);
+        when(polineTarget.queryParam(anyString(), any())).thenReturn(polineTarget);
+        when(polineTarget.request()).thenReturn(builder);
+        when(builder.accept(anyString())).thenReturn(builder);
+        when(builder.buildGet()).thenReturn(invocation);
+        PoLines mockPolines = new PoLines();
+        mockPolines.setTotalRecordCount(Integer.valueOf(2));
+        when(invocation.invoke((Class<Object>) any())).thenReturn(mockPolines);
+
+        AlmaPOLineService almaPOLineService = new AlmaPOLineService(mockAlmaApiClient);
+        String q = "q";
+        String status = "CLOSED";
```

Comment:
Legge dette i en variable evt. en constant?

Reply:
Gjelder for alle "magic-strings"

Reply:
Ok, da har jeg fjernet alt magi fra koden.